src/flb_log.c

```diff
@@ -62,10 +62,83 @@ static inline int consume_byte(flb_pipefd_t fd)
     return 0;
 }
 
+static inline int log_file_sz_max(off_t size, struct flb_log *log)
+{
+    double sz_mb;
+
+    sz_mb = size / (1024.00 * 1024.00);
+    if (sz_mb >= log->max_sz_mb) {
+        return FLB_TRUE;
+    }
+
+    return FLB_FALSE;
+}
+
+static inline void log_file_rotate(struct flb_log *log)
+{
+    char *backup_file_name = NULL;
+    char *src_file_name = NULL;
+    int ret = -1;
+    int i;
+
+    if (log->max_history == 0) {
+        /*
+         * no backups.
+         * remove the file since size has exceeded.
+         */
+        remove(log->out);
+        return;
+    }
+
+    /* max 2 digits for backup name suffix and 1 each for '.' and '\0' */
+    size_t filename_len = strlen(log->out) + 4;
+    backup_file_name = flb_malloc(filename_len);
+    if (!backup_file_name) {
+        flb_errno();
+        return;
+    }
+    src_file_name = flb_malloc(filename_len);
+    if (!src_file_name) {
+        flb_errno();
+        return;
+    }
+
+    for (i = log->max_history; i > 1; i--) {
```

Comment:
I think we might want to refactor this function, if we set the history limit to 200 it will try to rename 200 files each time which with this code structure is necessary to keep the code from messing up preexisting rotated files.